analytics/track.go

```diff
@@ -0,0 +1,80 @@
+package analytics
+
+import (
+	"bytes"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+const poolSize = 10
+const bufferSize = 100
+
+// Tracker ...
+type Tracker interface {
+	Enqueue(eventName string, properties map[string]interface{})
+	Wait()
+}
+
+type tracker struct {
+	jobs             chan *bytes.Buffer
+	waitGroup        *sync.WaitGroup
+	client           Client
+	sharedProperties map[string]interface{}
+}
+
+// NewDefaultTracker ...
+func NewDefaultTracker(shared map[string]interface{}) Tracker {
+	return NewTracker(make(chan *bytes.Buffer, bufferSize), &sync.WaitGroup{}, poolSize, NewDefaultClient(), shared)
+}
+
+// NewTracker ...
+func NewTracker(jobs chan *bytes.Buffer, waitGroup *sync.WaitGroup, size int, client Client, sharedProperties map[string]interface{}) Tracker {
```

Comment:
do we need to wire in anything else than `client` and `sharedProperties` ? think neither in production nor in the test will we alter these.

Reply:
NOTE: suggest unifying the var name of `sharedProperties` (I suggest `shared`) to clarify: both sets the same field of the struct.

Reply:
Ok I'll hardcode the pool mechanism